.github/workflows/deploy.yml

```diff
@@ -9,13 +10,29 @@ on:
       - '**/*.ts'
     branches:
       - main
+      - staging
 
 jobs:
-  if_merged:
+  deploy:
     if: github.event.pull_request.merged == true
     runs-on: ubuntu-latest
-
     steps:
+      - name: Determine environment
+        id: set-env
+        run: |
+          if [[ "${{ github.base_ref }}" == "main" ]]; then
+            echo "ENV=production" >> $GITHUB_ENV
+            echo "DEPLOY_PATH=${{ secrets.DEPLOY_PATH_PROD }}" >> $GITHUB_ENV
+            echo "PM2_NAME=bot-prod" >> $GITHUB_ENV
+          elif [[ "${{ github.base_ref }}" == "staging" ]]; then
+            echo "ENV=staging" >> $GITHUB_ENV
+            echo "DEPLOY_PATH=${{ secrets.DEPLOY_PATH_STAGING }}" >> $GITHUB_ENV
+            echo "PM2_NAME=bot-staging" >> $GITHUB_ENV
+          else
+            echo "Skipping deployment"
+            exit 1
+          fi
```

Comment:
if DEPLOY_PATH_PROD doesn't exists and it is a PR that is trying to merge in main branch, then we should skip prod deployment

if DEPLOY_PATH_STAGING doesn't exists and it is a PR that is trying to merge in staging branch, then we should skip deployment

Reply:
Left do it in another Pr, just for testing it works as it is currently